tests/ci/fast_test_check.py

```diff
@@ -42,14 +42,15 @@
 
 def get_fasttest_cmd(workspace, output_path, repo_path, pr_number, commit_sha, image):
     return (
-        f"docker run --cap-add=SYS_PTRACE "
+        f"docker run --cap-add=SYS_PTRACE --user={os.geteuid()}:{os.getegid()} "
```

Comment:
What for running under current user is required?

This breaks attaching with gdb:

```
Could not attach to process.  If your uid matches the uid of the target
2024-01-26 16:46:51 process, check the setting of /proc/sys/kernel/yama/ptrace_scope, or try
2024-01-26 16:46:51 again as the root user.  For more details, see /etc/sysctl.d/10-ptrace.conf
2024-01-26 16:46:51 ptrace: Inappropriate ioctl for device.
```

Since gdb is done from clickhouse-test which spawned by `run.sh`, while the server is spawed from `run.sh`, so server is not a child of the clickhouse-test and default YAMA policy forbid this.

https://s3.amazonaws.com/clickhouse-test-reports/59253/568a6f95d74c2708e062c37135ddb713eaf95b4a/fast_tests/run.log

Reply:
preventing spoilage of local files and playing with `chown` in the post process

I see. I've spent some time to work around it, and I think the least intrusive approach would be setting `chmod u+s /usr/bin/gdb` in Dockerfile.

The gdb is isolated there, and we actually want it to be overpowered, right?

![image](https://github.com/ClickHouse/ClickHouse/assets/3025537/a3309057-6484-41bc-9490-1557cc38d3ae)

Reply:
>The gdb is isolated there, and we actually want it to be overpowered

I guess this will be OK :)

Reply:
#59337